packaging/os/sorcery.py

```diff
@@ -0,0 +1,637 @@
+    update:
+        description:
+            - Whether or not to update sorcery scripts at the very first stage
+        required: false
+        choices: ["yes", "no"]
+        default: "no"
+
+    update_cache:
+        description:
+            - Whether or not to update grimoire collection before casting spells
+        required: false
+        aliases: ["update_codex"]
+        choices: ["yes", "no"]
+        default: "no"
+
+    cache_valid_time:
+        description:
+            - Time in seconds to invalidate grimoire collection on update
+            - especially useful for SCM and rsync grimoires
+            - makes sense only in pair with C(update_cache)
+        required: false
+        default: null
+'''
+
+
+EXAMPLES = '''
+# Make sure spell 'foo' is installed
+- sorcery: spell=foo state=present
+
+# Make sure spells 'foo', 'bar' and 'baz' are removed
+- sorcery: spell=foo,bar,baz state=absent
+
+# Make sure spell 'foo' with dependencies 'bar' and 'baz' is installed
+- sorcery: spell=foo depends=bar,baz state=present
+
+# Make sure spell 'foo' with 'bar' and without 'baz' dependencies is installed
+- sorcery: spell=foo depends=+bar,-baz state=present
+
+# Make sure spell 'foo' with libressl (providing SSL) dependency is installed
+- sorcery: spell=foo depends=libressl(SSL) state=present
+
+# Playbook: make sure spells with/without required dependencies (if any) are installed
+- sorcery: name={{ item.spell }} depends={{ item.depends | default(None) }} state=present
+  with_items:
+    - { spell: 'vifm', depends: '+file,-gtk+2' }
+    - { spell: 'fwknop', depends: 'gpgme' }
+    - { spell: 'pv,tnftp,tor' }
+
+# Install the latest version of spell 'foo' using regular glossary
+- sorcery: name=foo state=latest
+
+# Rebuild spell 'foo'
+- sorcery: spell=foo state=rebuild
+
+# Rebuild the whole system, but update Sorcery and Codex first
+- sorcery: spell=* state=rebuild update=yes update_cache=yes
+
+# Refresh the grimoire collection if it's 1 day old using native sorcerous alias
+- sorcery: update_codex=yes cache_valid_time=86400
+
+# Update only Sorcery itself
+- sorcery: update=yes
+'''
+
+
+RETURN = '''
+'''
+
+
+import datetime
+import fileinput
+import os
+import re
+import shutil
+
+
+# auto-filled at module init
+SORCERY = {
+    'sorcery': None,
+    'scribe': None,
+    'cast': None,
+    'dispel': None,
+    'gaze': None
+}
+
+SORCERY_LOG_DIR = "/var/log/sorcery"
+SORCERY_STATE_DIR = "/var/state/sorcery"
+
+
+def exec_command(command, module):
+    """ Run sorcery commands without asking questions.
+
+    This prevents commands from stuck due to queries and forces to use default
+    answers. It also leverages output if a user has the VOYEUR option turned on
+    in Sorcery.
+
+    """
+
+    prompt_env = "env PROMPT_DELAY=0 VOYEUR=0"
+
+    return module.run_command("%s %s" % (prompt_env, command))
+
+
+def get_sorcery_ver(module):
+    """ Get Sorcery version. """
+
+    cmd_sorcery = "%s --version" % SORCERY['sorcery']
```

Comment:
for this and other command line tools, you should rely on `module.get_bin_path` and not just that the user's path is correctly set.